Documentation/networking/dns.md

```diff
@@ -9,25 +21,91 @@ $ sudo rkt run --dns=8.8.8.8 pod.aci
 Other parameters can be given:
 
 ```bash
-$ sudo rkt run --dns=8.8.8.8 --dns=8.8.4.4 --dns-search=foo.com --dns-opt=debug pod.aci
+$ sudo rkt run \
+	--dns=8.8.8.8 --dns=4.2.2.2 \
+	--dns-domain=example.org \
+	--dns-opt=debug --dns-opt=rotate \
+	--dns-search=example.com --dns-search=example.gov \
+	pod.aci
 ```
 
 This will generate the following `/etc/resolv.conf` for the applications:
 
 ```
-# Generated by rkt
+# Generated by rkt run
 
-search foo.com
+search example.com example.gov
 nameserver 8.8.8.8
-nameserver 8.8.4.4
-options debug
+nameserver 4.2.2.2
+options debug rotate
+domain example.org
 ```
 
+### "Magic" parameters
+
+#### `host`
+The magic parameter `host` will bind-mount the host's `/etc/resolv.conf` in to the applications.
+This will be a read-only mount.
+
+#### `none`
+The magic parameter `none` will ignore any DNS configuration from CNI. This will ensure that
+the image's `/etc/resolv.conf` has precedence.
+
+### Precedence
+`resolv.conf` can be generated by multiple components. The order of precedence is:
+
+1. If `--dns`, et al. are passed to `rkt run`
+2. If a CNI plugin returns DNS information, unless `--dns=none` is passed
+3. If a volume is mounted on `/etc/resolv.conf`
+4. If the application container includes `/etc/resolv.conf`
+
+
+
+## `/etc/hosts`
```

Comment:
I think this section should document a bit better the "some default localhost entries" in the example below. What did we settle on in the end? Does this include localname + fqdn attached to 127.0.0.1?